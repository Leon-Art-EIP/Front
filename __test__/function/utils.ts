import type { ReactTestRendererJSON } from "react-test-renderer";
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from "react-test-renderer";
import { expect } from "vitest";

function toJson(component: renderer.ReactTestRenderer): ReactTestRendererJSON {
  const result = component.toJSON();
  expect(result).toBeDefined();

  return result as ReactTestRendererJSON;
}

export function testSnapshot(element: JSX.Element): void {
  const component = renderer.create(element);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
}
