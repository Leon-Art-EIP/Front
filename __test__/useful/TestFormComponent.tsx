import { FormProvider } from "react-hook-form";
import type { ReactNode } from "react";
import useTestForm from "./useTestForm";

interface ITestInputProps {
  name: string;
  value?: string[] | string;
  children: ReactNode;
}

export default function TestFormComponent(props: ITestInputProps): JSX.Element {
  const methods = useTestForm({ [props.name]: props.value });

  return <FormProvider {...methods}>{props.children}</FormProvider>;
}
