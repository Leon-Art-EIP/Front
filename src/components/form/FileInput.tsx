import { AddSharp } from "@mui/icons-material";
import { useController } from "react-hook-form";
import { cn } from "../../tools/cn";

interface IFileInputProps {
  title?: string;
  name: string;
  className?: string;
}

export default function FileInput(props: IFileInputProps): JSX.Element {
  const {
    field: { onChange, value },
    fieldState: { error },
  } = useController({ name: props.name });

  return (
    <div className={cn("flex flex-col gap-2", props.className)}>
      <div className="text-lg">{props.title}</div>
      <label
        htmlFor="file_input"
        className="hover:cursor-pointer bg-secondaryGrey rounded flex justify-center items-center h-24 text-gray-400"
      >
        <AddSharp style={{ fontSize: 50 }} />
      </label>
      <input id="file_input" type="file" className="hidden" value={value} onChange={onChange} />
      <p>{value.split("\\").pop()}</p>
      {error && <div className="text-red-600">{error.message}</div>}
    </div>
  );
}
