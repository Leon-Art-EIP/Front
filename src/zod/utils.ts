export const getValueOrUndefined = (arg: unknown) =>
  arg === "" || arg === undefined || arg === null ? undefined : arg;
