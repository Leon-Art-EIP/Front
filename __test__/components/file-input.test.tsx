import { test } from "vitest";
import { testSnapshot } from "../function/utils";
import TestFormComponent from "../useful/TestFormComponent";
import FileInput from "../../src/components/form/FileInput";

test("FileInput snapshot", () => {
  testSnapshot(
    <TestFormComponent name="file">
      <FileInput name="file" />
    </TestFormComponent>
  );
});
