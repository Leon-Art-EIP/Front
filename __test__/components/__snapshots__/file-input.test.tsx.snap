// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FileInput snapshot 1`] = `
<div
  className="flex flex-col gap-2"
>
  <button
    className="hover:cursor-pointer bg-secondary rounded flex justify-center items-center text-tertiary h-full"
    onClick={[Function]}
    onDragLeave={[Function]}
    onDragOver={[Function]}
    onDrop={[Function]}
    title="Importer"
    type="button"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="AddSharpIcon"
      focusable="false"
      style={
        {
          "fontSize": 50,
        }
      }
      viewBox="0 0 24 24"
    >
      <path
        d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
      />
    </svg>
  </button>
  <input
    accept="image/*"
    className="hidden"
    id="file"
    multiple={false}
    name="file"
    onChange={[Function]}
    type="file"
  />
</div>
`;
